### PR TITLE
journal: Add more information to --verify error messages

### PR DESCRIPTION
**Pull request by daandemeyer**
Includes #21306 (only the last commit is new)

### CODE REVIEW

File: src/libsystemd/sd-journal/journal-verify.c [hunk 1]
```diff
@@ -1009,7 +1017,7 @@ int journal_file_verify(
                 if (!!(o->object.flags & OBJECT_COMPRESSED_XZ) +
                     !!(o->object.flags & OBJECT_COMPRESSED_LZ4) +
                     !!(o->object.flags & OBJECT_COMPRESSED_ZSTD) > 1) {
-                        error(p, "Object has multiple compression flags set");
+                        error(p, "Object has multiple compression flags set (flags: %u)", o->object.flags);
```
Comment (yuwata) [inline src/libsystemd/sd-journal/journal-verify.c]: Maybe, in hex is better for printing flags?

**Review by yuwata — APPROVED** [review]
LGTM. One minor suggestion.

Also, #21306 was already merged. Please rebase.

File: src/libsystemd/sd-journal/journal-verify.c [hunk 2]
```diff
@@ -1008,7 +1016,7 @@ int journal_file_verify(
                 if (!!(o->object.flags & OBJECT_COMPRESSED_XZ) +
                     !!(o->object.flags & OBJECT_COMPRESSED_LZ4) +
                     !!(o->object.flags & OBJECT_COMPRESSED_ZSTD) > 1) {
-                        error(p, "Object has multiple compression flags set");
+                        error(p, "Object has multiple compression flags set (flags: %x)", o->object.flags);
```
Comment (poettering) [inline src/libsystemd/sd-journal/journal-verify.c]: `%x` → `0x%x`

**Comment by poettering** [comment]
lgtm, just one nit

**Comment by DaanDeMeyer** [comment]
Fixed nit, upgrading green label

**Pull request merged**